crates/core/tedge_api/Cargo.toml

```diff
@@ -9,25 +9,35 @@ rust-version = "1.58.1"
 
 [dependencies]
 clock = { path = "../../common/clock" }
+csv = "1.1"
+download = { path = "../../common/download" }
+futures = "0.3"
+futures-util = "0.3"
 json-writer = { path = "../../common/json_writer" }
+hyper = { version = "0.14", features = ["full"] }
 mqtt_channel = { path = "../../common/mqtt_channel" }
+nanoid = "0.4"
+path-clean = "0.1"
+routerify = "3.0"
```

Comment:
I'm surprised to see so many new dependencies. This raises a question: are we just moving the API there or are we also moving the mechanisms? If so it would be good to add a feature flag so only the agent will include the mechanisms.

Reply:
I thought about feature flags too. I will work on adding feature flags to split again the two merged crates.